debezium-embedded/src/main/java/io/debezium/embedded/AsyncEngineConfig.java

```diff
@@ -0,0 +1,65 @@
+ */
+package io.debezium.embedded;
+
+import io.debezium.config.Field;
+
+/**
+ * Configuration options specific to {@link AsyncEmbeddedEngine}.
+ *
+ * @author vjuranek
+ */
+public interface AsyncEngineConfig extends EmbeddedEngineConfig {
+
+    int AVAILABLE_CORES = Runtime.getRuntime().availableProcessors();
+
+    /**
+     * An optional field that specifies the maximum amount of time to wait for task lifecycle operation, i.e. for starting and stopping the task.
+     */
+    Field TASK_MANAGEMENT_TIMEOUT_MS = Field.create("task.management.timeout.ms")
+            .withDescription("Time to wait for task's lifecycle management operations (starting and stopping), given in milliseconds. "
+                    + "Defaults to 5 seconds (5000 ms).")
+            .withDefault(5_000L)
+            .withValidation(Field::isPositiveInteger);
+
+    /**
+     * An optional field that specifies the number of threads to be used for processing CDC records.
+     */
+    Field RECORD_PROCESSING_THREADS = Field.create("record.processing.threads")
+            .withDescription("The number of threads to be used for processing CDC records. The default is number of available machine cores.")
+            .withDefault(AVAILABLE_CORES)
+            .withValidation(Field::isPositiveInteger);
+
+    /**
+     * An optional field that specifies maximum time in ms to wait for submitted records to finish processing when the task shut down is called.
+     */
+    Field RECORD_PROCESSING_SHUTDOWN_TIMEOUT_MS = Field.create("record.processing.shutdown.timeout.ms")
+            .withDescription("Maximum time in milliseconds to wait for processing submitted records when task shutdown is called. The default is 10 seconds (10000 ms).")
+            .withDefault(1000L)
+            .withValidation(Field::isPositiveInteger);
+
+    /**
+     * An optional field that specifies how the records will be produced. Sequential processing (the default) means that the records will be produced in the same order
+     * as the engine obtained them from the connector. Non-sequential processing means that the records can be produced in arbitrary order, typically once the record is
+     * transformed and/or serialized.
+     */
+    Field RECORD_PROCESSING_SEQUENTIALLY = Field.create("record.processing.sequentially")
+            .withDescription("Determines how the records should be produce. Sequential processing means (setting to `true`, the default) that the records are "
+                    + "produced in the same order as they were obtained from the database. Non-sequential processing means that the records can be produced in a different order "
+                    + "order than the original one, but this approach given better throughput, as the records are produced immediately once the SMTs and serialization of "
+                    + "the message is done, without waiting of other records.")
```

Comment:
What's the use-case for non-sequential aka "unordered" processing?

Reply:
Well, there were users who are interested only in knowing that event happened, not about order or payload and require high performance. Yet I'd recommend to change this option a bit. It should be named `record.processing.order` with values `ordered` and `unordered` and in future a new option will be added `ordered_per_key`.
@vjuranek WDYT?

Reply:
@rk3rn3r you may take a look on the [design document](https://github.com/debezium/debezium-design-documents/pull/8) first to get some explanation what this implementation intend to do and how.

@jpechane yes, definitely make sense. I expect this discussion happen under https://github.com/debezium/debezium-design-documents/pull/8 This is what I meant yesterday when I said the design document should be reviewed first. No problem to discussed it here, but I'd like not to exclude other people (like Jeremy, who originally came with this idea) from the discussion.

Reply:
The third option is out of scope of this work, but I want to have config ready so we don't need than to play with config options again.

Reply:
done

Reply:
In ideal case the parallelism can be aligned to the number of partitions for sinks that use partitioned storage like Kafka with its partitions. Parallelism could be the number of (active) partitions. But it will only work for downstream systems that support and expose partitioning and the used algorithm for partitioning. Aligning this might be too complex for systems where those parameters are known and impossible for systems that hides those information.